src/components/BeamContext.tsx

```diff
@@ -40,24 +41,25 @@ export function BeamProvider({ children }: { children: ReactNode }) {
   const [, tick] = useReducer((prev) => prev + 1, 0);
   const modalRef = useRef<ModalProps | undefined>();
   const modalBodyDiv = useMemo(() => document.createElement("div"), []);
+  const modalCanCloseChecksRef = useRef<CheckFn[]>([]);
   const modalFooterDiv = useMemo(() => document.createElement("div"), []);
-  const contentStackRef = useRef<ContentStack[]>([]);
-  const canCloseModalChecksRef = useRef<Array<() => boolean>>([]);
-  const canCloseDrawerChecksRef = useRef<Array<() => boolean>>([]);
-  const canCloseDrawerDetailsChecksRef = useRef<Array<Array<() => boolean>>>([]);
+  const drawerContentStackRef = useRef<ContentStack[]>([]);
+  const drawerCanCloseChecks = useRef<CheckFn[]>([]);
+  const drawerCanCloseDetailsChecks = useRef<CheckFn[][]>([]);
 
   // We essentially expose the refs, but with our own getters/setters so that we can
   // have the setters call `tick` to re-render this Provider
   const context = useMemo<BeamContextState>(() => {
     return {
+      // These two keys need to trigger re-renders on change
       modalState: new PretendRefThatTicks(modalRef, tick),
-      contentStack: new PretendRefThatTicks(contentStackRef, tick),
-      // We don't need to rerender when these are mutated, so just expose as-is
+      drawerContentStack: new PretendRefThatTicks(drawerContentStackRef, tick),
+      // The rest we don't need to re-render when these are mutated, so just expose as-is
+      modalCanCloseChecks: modalCanCloseChecksRef,
       modalBodyDiv,
       modalFooterDiv,
-      canCloseModalChecks: canCloseModalChecksRef,
-      canCloseDrawerChecks: canCloseDrawerChecksRef,
-      canCloseDrawerDetailsChecks: canCloseDrawerDetailsChecksRef,
+      drawerCanCloseChecks,
+      drawerCanCloseDetailsChecks,
```

Comment:
Will there be any external breaking changes?

Reply:
Good question; I'm pretty sure no